### PR TITLE
reduce garbage for repeated `zig cc -` calls

### PR DESCRIPTION
**Pull request by motiejus**
Context: user provides `-` to read from stdin instead of a file.

Before: it would create a file in $ZIG_LOCAL_CACHE_DIR/tmp/ for each invocation and leave it there.

After: it hashes the file contents and renames the file to the hash.

Result: repeated invocations of `zig cc` do not cause so much trash to be created.

### CODE REVIEW

**Comment by andrewrk** [comment]
Sorry for the delay - I should have merged this a long time ago. I was sad to see the `writeFileAll` call go, but this is perfectly fine.

File: src/main.zig [hunk 1]
```diff
@@ -3129,19 +3129,36 @@ fn buildOutputType(
...
+        // `zig cc -` will result in the same temp file name.
+        var f = try local_cache_directory.handle.createFile(dump_path, .{});
         defer f.close();
-        try f.writeFileAll(io.getStdIn(), .{});
+
+        // Re-using the hasher from Cache, since the functional requirements
+        // for the hashing algorithm here and in the cache are the same.
+        // We are providing our own cache key, because this file has nothing
+        // to do with the cache manifest.
+        var hasher = Cache.Hasher.init("0123456789abcdef");
```
Comment (mlugg) [inline src/main.zig:3149]: Should this not be random?

Comment (andrewrk) [reply src/main.zig:3149]: the purpose is for the file path to be a hash of the contents

Comment (mlugg) [reply src/main.zig:3149]: Yes, I know, but the string passed here is a seed, right? So it ought to be a hardcoded string of *random* bytes, like what `std.Build.Cache.hasher_init` does.

Comment (andrewrk) [reply src/main.zig:3149]: random bytes, all zeroes, 123abc, it really doesn't matter. I think best would be using `Cache.hasher_init`.

**Pull request merged**